ProcessOptimizer/plots.py

```diff
@@ -1,2382 +1,2493 @@
+    """
+    # The idea is to step through one dimension and evaluating the model with
+    # that dimension fixed.
+    # (Or step through 2 dimensions when i and j are given.)
+    # Categorical dimensions make this interesting, because they are one-
+    # hot-encoded, so there is a one-to-many mapping of input dimensions
+    # to transformed (model) dimensions.
+
+    # If we havent parsed an x_eval list we use random sampled values instead
+    if x_eval is None:
+        sample_points = space.transform(space.rvs(n_samples=n_samples))
+    else:
+        sample_points = space.transform([x_eval])
+
+    # dim_locs[i] is the (column index of the) start of dim i in sample_points.
+    # This is usefull when we are using one hot encoding,
+    # i.e using categorical values.
+    dim_locs = np.cumsum([0] + [d.transformed_size for d in space.dimensions])
+
+    if j is None:
+        # We sample evenly instead of randomly. This is necessary when using
+        # categorical values
+        xi, xi_transformed = space.dimensions[i].evenly_sample(n_points)
+        yi = []
+        stddevs = []
+        for x_ in xi_transformed:
+            rvs_ = np.array(sample_points)  # copy
+            # We replace the values in the dimension that we want to keep fixed
+            rvs_[:, dim_locs[i]: dim_locs[i + 1]] = x_
+            # In case of `x_eval=None` rvs conists of random samples.
+            # Calculating the mean of these samples is how partial dependence
+            # is implemented.
+            funcvalue, stddev = model.predict(rvs_, return_std=True)
+            yi.append(np.mean(funcvalue))
+            stddevs.append(np.mean(stddev))
+        # Convert yi and stddevs from lists to numpy arrays
+        yi = np.array(yi)
+        stddevs = np.array(stddevs)
+
+        return xi, yi, stddevs
+
+    else:
+        xi, xi_transformed = space.dimensions[j].evenly_sample(n_points)
+        yi, yi_transformed = space.dimensions[i].evenly_sample(n_points)
+
+        # stddev structure is made regardless of whether it is returned,
+        # since this is cheap and makes the code simpler.
+        zi = []
+        stddev_matrix = []
+        for x_ in xi_transformed:
+            value_row = []
+            stddev_row = []
+            for y_ in yi_transformed:
+                rvs_ = np.array(sample_points)  # copy
+                rvs_[:, dim_locs[j]: dim_locs[j + 1]] = x_
+                rvs_[:, dim_locs[i]: dim_locs[i + 1]] = y_
+                funcvalue, stddev = model.predict(rvs_, return_std=True)
+                value_row.append(np.mean(funcvalue))
+                stddev_row.append(np.mean(stddev))
+            zi.append(value_row)
+            stddev_matrix.append(stddev_row)
+        if return_std:
+            return xi, yi, np.array(zi).T, np.array(stddev_matrix).T
+        else:
+            return xi, yi, np.array(zi).T
+
+
+def plot_objective(
+    result,
+    levels=10,
+    oversampling=None,
```

Comment:
Do we really need to bloat the function call with user input on oversampling? I think we should develop good settings that simply live inside the function instead of allowing user input on this. We don't provide handles for changing aestetic aspects of the plots (color of points etc.) for the same reason.